CONTRIBUTING.md

```diff
@@ -0,0 +1,368 @@
+Studio / the Android SDK.
+
+Configure your environment, so that the `ANDROID_HOME` and `JAVA_HOME`
+environment variables are set. A convenient way of doing this is to add them
+into `~/.bashrc`. On a Mac, the SDK and Java installation used by the SDK may
+be found:
+
+```
+export ANDROID_HOME=~/Library/Android/sdk
+export JAVA_HOME=/Applications/Android\ Studio.app/Contents/jre/jdk/Contents/Home
+```
+Note: JDK 11, 12, 13, etc. have known issues and are not supported.
+
+Now, clone the AWS SDK for Android project from GitHub.
+
+```
+git clone git@github.com:aws-amplify/aws-sdk-android.git
+```
+Load this project into Android Studio by selecting File > Open, and choosing
+the root directory of the project (`aws-sdk-android`). Alternately, cd into this
+top-level directory.
+
+In Android Studio, build the project by clicking the Hammer icon, "Make Project
+⌘F9". If working on the command line, you can do the same thing via:
+
+```
+./gradlew build
+```
+
+## Tools
+[Gradle](https://gradle.org) is used for all [build and dependency management](https://developer.android.com/studio/build).
+
+Some widely used dependencies are:
+
+1. [org.json](https://developer.android.com/reference/org/json/JSONObject) is
+   baked into Android, and is used for all modeling with JSON.
+2. [Gson](https://github.com/google/gson/blob/master/README.md#gson) is used
+   for serialization and deserialization.
+3. [OkHttp](https://github.com/square/okhttp#okhttp) is used for network operations.
+
+_Unit and component tests_, which run on your development machine, use:
+
+1. [jUnit](https://github.com/junit-team/junit4/blob/r4.13/README.md#junit-4), to make assertions
+2. [Mockito](https://javadoc.io/static/org.mockito/mockito-core/1.10.19/org/mockito/Mockito.html#1), to mock dependencies
+3. [Robolectric](https://github.com/robolectric/robolectric/blob/master/README.md),
+   to simulate an Android device when unit tests execute on your machine.
+
+_Instrumentation tests_, which run on an Android device or emulator use
+AndroidX test core, runner, and a jUnit extension. See Android's notes on
+[using AndroidX for test](https://developer.android.com/training/testing/set-up-project).
+
+## SDK Fundamentals
+These are only a few fundamentals that are helpful to know when developing against the AWS SDK for Android.
+
+* Never embed credentials in an Android application.  It is trivially easy to decompile applications and steal embedded credentials.  Always use temporarily vended credentials from services such as Amazon Cognito Identity.
+* Unless explicitly stated, calls are synchronous and must be taken off of the main thread.
+* Unless explicitly stated, calls can always throw an AmazonServiceException or an AmazonClientException (depending on if the exception is generated by the service or the client respectively).
+* The SDK will handle re-trying requests automatically, but unless explicitly stated will throw an exception if it cannot contact AWS.
```

Comment:
Wording -- I suggest, 
```
The SDK will automatically retry requests. Unless explicitly stated otherwise, a call with throw an exception when all retries have been exhuasted.
```

Reply:
done